### PR TITLE
Added platform check - debian and rhel

### PR DESCRIPTION
**Pull request by grdnrio**
Wrapped default in platform family case for debian and rhel. This allows this hardening cookbook to be added to a remediation role that can be used to address multiple platforms.

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -19,12 +19,17 @@
...
-include_recipe('os-hardening::limits')
-include_recipe('os-hardening::login_defs')
-include_recipe('os-hardening::minimize_access')
-include_recipe('os-hardening::pam')
-include_recipe('os-hardening::profile')
-include_recipe('os-hardening::securetty')
-include_recipe('os-hardening::suid_sgid') if node['os-hardening']['security']['suid_sgid']['enforce']
-include_recipe('os-hardening::sysctl')
+case node['platform_family']
+when 'debian', 'rhel'
```
Comment (artem-sidorenko) [inline recipes/default.rb]: @grdnrio in my eyes a such thing might lead to confusion: as a cookbook user I usually do not expect such logic in the default recipe of included cookbook, I expect that the included code gets executed. If such `if` is required, it belongs in my eyes to some wrapper cookbook but not here

**Comment by chris-rock** [comment]
The reason behind @grdnrio PR is the ability to add e.g. `os-hardening` and `windows-hardening` in a simple role and assign the role to all environments, so that basic hardening is ensured across the server fleet. Personally, I do not like that we duplicate information here, since we add the supported OS in our `metadata.rb`. 

By not having a guard, we may execute the hardening on a non-supported system (e.g. unix). Could we break something here?

I asked for the best-practice guidance in https://chefcommunity.slack.com/archives/C47C2PJF9/p1490084832443358 As soon as I know more, I'll add the result in here.

**Comment by artem-sidorenko** [comment]
@chris-rock @grdnrio this use case makes definitely sense. What about the following idea?

- we still have cookbooks like `chef-windows-hardening` and `chef-linux-hardening` (currently its `chef-os-hardening`)
- we have a wrapper cookbook `chef-os-hardening` (it does not exist yet) with `chef-windows-hardening` and `chef-linux-hardening` as dependencies and logic like this:

```ruby
case node['os']
when 'linux'
  include_recipe 'linux-hardening'
when 'windows'
  include_recipe 'windows-hardening'
else
  raise "Unsupported OS #{node['os']}"
end
```

Yes, we will need some kind of transition because of cookbook renaming. In my eyes this would be a better solution that having such guards in the cookbooks:
 - users who need only linux/windows hardening logic can choose the right cookbook without to pull other not required dependencies
 - users who need a mixed os-hardening cookbook/role, can choose the new wrapper cookbook
 - we do not have such skip guards in the particular cookbooks: e.g. if user includes directly the linux-hardening - it runs or fails, but it is not skipped silently

**Comment by chris-rock** [comment]
screen shot 2017-03-21 at 5 30 17 pm

**Comment by chris-rock** [comment]
@artem-sidorenko I agree that chef-linux-hardening would be a good naming too. Currently I am not happy with the solution to create a wrapper cookbook that just includes 8 lines of code (and is unlikely to change in future)

**Comment by chris-rock** [comment]
@arlimus @atomic111 could you add your thoughts here?

**Comment by artem-sidorenko** [comment]
> Currently I am not happy with the solution to create a wrapper cookbook that just includes 8 lines of code (and is unlikely to change in future)

@chris-rock I understand this completely

My major pain point is: I do not like a solution in the security area which can silently stop to run (I already saw situations where ohai got wrong data in the automatic attributes, because of some bugs in the distros). I do not think its a good tradeoff from security point of view: to include here this logic, but to have the risk of silent failure (not everybody has chef compliance with continuously running inspec tests to notice a such thing)

**Comment by grdnrio** [comment]
Yes it seems like the wrapper is the solution, but my aim was to match the InSpec meta profile behaviour of platform agnosticism in a single asset. InSpec decides automatically though, so is smarter :) It's more of a Chef problem.

The answer is probably a wrapper cookbook instead of a role, but it feels like bad behaviour to be using roles and (role) wrapper cookbooks to achieve the same thing. Not something to be solved here I guess!

**Pull request closed without merging**